testing_dir/minesweeper.py

```diff
@@ -89,7 +89,7 @@ def game_loop():
             player_pos = new_pos
             moves += 1
 
-        if moves == BOARD_SIZE * BOARD_SIZE - NUM_MINES:
+        if moves == BOARD_SIZE * BOARD_SIZE - NUM_MINES + 1:  # Error: should not have +1
```

Comment:
Hello!

The issue with including `+1` in the condition `if moves == BOARD_SIZE * BOARD_SIZE - NUM_MINES + 1:` is that it could cause the game to end one move earlier than intended. The condition is meant to check if the player has made the exact number of moves required to uncover all non-mine cells. By adding `+1`, the game might incorrectly conclude that all mines have been found when there is still one more move needed. This could lead to a premature game termination, preventing the player from actually completing the game.

Here's the corrected line:

```python
if moves == BOARD_SIZE * BOARD_SIZE - NUM_MINES:
```

This ensures the game only ends when all non-mine cells are uncovered.

> Response by EZPR_Bot